### ISSUE TITLE
I'm having trouble using Express ErrorHandler with ZodiosContext

### ISSUE THREAD

**Issue by thelinuxlich** [ecyrbe/zodios-express#80]
app in this example is a `ctx.app(api)`

![image](https://user-images.githubusercontent.com/5043/204383356-79ff37d1-e090-4062-bdfe-810de09bc58f.png)

Full error:

```
[{
	"resource": "/home/thelinuxlich/onepay/1p/packages/api/src/index.ts",
	"owner": "typescript",
	"code": "2769",
	"severity": 8,
	"message": "No overload matches this call.\n  Overload 1 of 4, '(...handlers: ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]): Omit<...> & ZodiosHandlers<...>', gave the following error.\n    Argument of type 'ErrorRequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>'.\n  Overload 2 of 4, '(handlers: ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]): Omit<...> & ZodiosHandlers<...>', gave the following error.\n    Argument of type 'ErrorRequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]'.\n  Overload 3 of 4, '(path: string, ...handlers: ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]): Omit<...> & ZodiosHandlers<...>', gave the following error.\n    Argument of type 'ErrorRequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'string'.",
	"source": "ts",
	"startLineNumber": 230,
	"startColumn": 9,
	"endLineNumber": 230,
	"endColumn": 21
}]
```

**Comment by ecyrbe** [comment]
Yes, error handler need to be used inline to take advantage of typesafety.

**Comment by thelinuxlich** [comment]
Doesn't work inline too 
![image](https://user-images.githubusercontent.com/5043/204401179-d25d2c58-69e4-47e0-bff5-08f23663a90a.png)

**Comment by ecyrbe** [comment]
if you remove the type of the parameters and let inference do it's job ?

**Comment by thelinuxlich** [comment]
Same thing 

![image](https://user-images.githubusercontent.com/5043/204402868-8c764e9f-0b11-4796-81f9-4db0823d202d.png)

**Comment by ecyrbe** [comment]
thank you for trying. i'll take a look into it.

**Comment by thelinuxlich** [comment]
Now thinking about this, if we aren't supposed to type the parameters how would it work having the routes in separate files?

Example: `app.get("/transactions", transactionsGet) // how transactionsGet can have typesafety on its file?`

**Comment by ecyrbe** [comment]
you declare a router and share it where you want to declare your get impl. But always inline the get handler. it's the same with tRPC. everything is declared inline to have typesafety and not redeclare types manually.

**Comment by thelinuxlich** [comment]
I'm also getting this weird error, but it doesn't stop `app` from typechecking

![image](https://user-images.githubusercontent.com/5043/204409107-9a642e0a-0886-4315-abc5-e303e6c722b0.png)

Full error:

```
[{
	"resource": "/home/thelinuxlich/onepay/1p/packages/api/src/index.ts",
	"owner": "typescript",
	"code": "2742",
	"severity": 8,
	"message": "The inferred type of 'app' cannot be named without a reference to '.pnpm/@types+express-serve-static-core@4.17.30/node_modules/@types/express-serve-static-core'. This is likely not portable. A type annotation is necessary.",
	"source": "ts",
	"startLineNumber": 30,
	"startColumn": 7,
	"endLineNumber": 30,
	"endColumn": 10
}]
```

**Comment by ecyrbe** [comment]
can you show your `tsconfig.json` file ?

**Comment by thelinuxlich** [comment]
here it is:

```json
{
  "compilerOptions": {
    "target": "esnext",
    "composite": true,
    "incremental": true,
    "module": "esnext",
    "moduleResolution": "node",
    "noEmit": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "exactOptionalPropertyTypes": false,
    "allowUnreachableCode": false,
    "isolatedModules": true,
    "noUncheckedIndexedAccess": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true,
    "preserveSymlinks": false
  },
  "include": ["src/**/*", "vitest.config.ts"],
  "exclude": ["node_modules"]
}
```

**Comment by thelinuxlich** [comment]
Interesting, now that I looked the tsconfig, I changed the preserveSymlinks option to true and the error vanished(only remaining the errorHandler problem), but I don't know the effects of setting this flag to true, should it be recommended on the docs?

**Comment by ecyrbe** [comment]
i think the default is for it to be false. So i'll try to reproducce, diagnose and fix the issue. 
The only option i want to make mandatory is `strict: true` .
i'll continue this tomorrow. thank you for the feedback loop.

**Comment by thelinuxlich** [comment]
Yeah, I just noticed that when set to true it doesn't infer the app.use parameters

**Comment by thelinuxlich** [comment]
I also noticed that `req.headers` doesn't get the headers from the API definition

**Comment by blade2005** [comment]
I'm encountering this same issue when trying to use express winston error handling.

```
const consoleTransport = new winston.transports.Console();

const winstonFormat = winston.format.combine(
  winston.format.colorize(),
  winston.format.json()
);

const loggingConfig: winston.LoggerOptions = {
  transports: [consoleTransport],
  format: winstonFormat
};
export const logger = winston.createLogger(loggingConfig);
const errorLoggingConfig: expressWinston.ErrorLoggerOptions = {
  winstonInstance: logger
};
export const winstonErrorLogger =
  expressWinston.errorLogger(errorLoggingConfig);
app.use(winstonErrorLogger);
```

The error thrown is

```
[{
	"resource": "/home/cdavis/Projects/dart/zodios-poc/src/app.ts",
	"owner": "typescript",
	"code": "2769",
	"severity": 8,
	"message": "No overload matches this call.\n  Overload 1 of 4, '(...handlers: ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]): Omit<...> & ZodiosHandlers<...>', gave the following error.\n    Argument of type 'ErrorRequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>'.\n  Overload 2 of 4, '(handlers: ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]): Omit<...> & ZodiosHandlers<...>', gave the following error.\n    Argument of type 'ErrorRequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]'.\n  Overload 3 of 4, '(path: string, ...handlers: ZodiosRouterContextRequestHandler<ZodObject<{ user: ZodObject<{ id: ZodNumber; name: ZodString; isAdmin: ZodBoolean; }, \"strip\", ZodTypeAny, { id: number; name: string; isAdmin: boolean; }, { ...; }>; }, \"strip\", ZodTypeAny, { ...; }, { ...; }>>[]): Omit<...> & ZodiosHandlers<...>', gave the following error.\n    Argument of type 'ErrorRequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' is not assignable to parameter of type 'string'.",
	"source": "ts",
	"startLineNumber": 20,
	"startColumn": 9,
	"endLineNumber": 20,
	"endColumn": 27
}]
```

In this case I don't see how I could get type inference to work.

**Comment by ecyrbe** [comment]
Hello,

This should be fixed on version `10.4.5` of `@zodios/express`, thank you the reports.
